clang/test/CheckedC/parsing/invalid-where-clause.c

```diff
@@ -36,3 +36,13 @@ void invalid_cases(_Nt_array_ptr<char> p, int a, int b) {
   _Where a++ < 1; // expected-error {{increment expression not allowed in expression}}
   _Where a _And p : _And q : count(0) _And x _And a = 1 _And f() < 0 _And; // expected-error {{invalid expression in where clause, expected bounds declaration or equality expression}} expected-error {{expected bounds expression}} expected-error {{use of undeclared identifier q}} expected-error {{use of undeclared identifier 'x'}} expected-error {{expected comparison operator in equality expression}} expected-error {{call expression not allowed in expression}} expected-error {{expected bounds declaration or equality expression in where clause}}
 }
+
+void f1(int a _Where a, _Nt_array_ptr<int> p : count(0) _Where p :); // expected-error {{invalid expression in where clause, expected bounds declaration or equality expression}} expected-error {{expected bounds expression}}
+
+void f2(int *p : itype(_Ptr<int>) _Where p, int n); // expected-error {{invalid expression in where clause, expected bounds declaration or equality expression}}
+
+void f3(int a _Where a == 1, a == 1); // expected-error {{unknown type name 'a'}} expected-error {{expected ')'}} expected-note {{to match this '('}}
+
+void f4(int a _Where a == 1, b == 1); // expected-error {{unknown type name 'b'}} expected-error {{expected ')'}} expected-note {{to match this '('}}
+
+void f5(int a _Where a,a); // expected-error {{redefinition of parameter 'a'}} expected-error {{invalid expression in where clause, expected bounds declaration or equality expression}} expected-warning {{type specifier missing, defaults to 'int'}} expected-note {{previous declaration is here}}
```

Comment:
Add negative test cases for:
1. Unbalanced parentheses.
2. Where clause occurring w/o any parameters in a function decl.